geonode/services/migrations/0048_service_harvester.py

```diff
@@ -0,0 +1,48 @@
+# Generated by Django 3.2.4 on 2021-09-20 13:10
+
+from django.db import migrations, models, transaction
+import django.db.models.deletion
+import logging
+logger = logging.getLogger(__name__)
+
+
+@transaction.atomic
+def set_harvester(apps, _):
+    from geonode.services import enumerations
+    from geonode.harvesting.models import Harvester
+    from geonode.harvesting.tasks import update_harvestable_resources
+
+    try:
+        MyModel = apps.get_model('services', 'Service')
+        _services_with_empty_harvesters = MyModel.objects.filter(
+            type__in=(enumerations.WMS, enumerations.GN_WMS),
+            harvester__isnull=True)
+        for _service in _services_with_empty_harvesters:
+            service_harvester = Harvester.objects.create(
+                name=_service.name,
+                default_owner=_service.owner,
+                remote_url=_service.service_url,
+                harvester_type=enumerations.HARVESTER_TYPES[_service.type]
+            )
+            service_harvester.update_availability()
```

Comment:
Let's avoid triggering side effects inside migrations. I would skip checking the availability and the retrieval of harvestable resources from the migration istelf.
Alignment of existing remote services will be done with a dedicated management command in case.